util/parser/ast.go

```diff
@@ -49,8 +49,76 @@ func (i *implicitDatabase) Leave(in ast.Node) (out ast.Node, ok bool) {
 	return in, true
 }
 
+func findTablePos(s, t string) int {
+	l := 0
+	for i := range s {
+		if s[i] == ' ' || s[i] == ',' {
+			if len(t) == i-l && strings.Compare(s[l:i], t) == 0 {
+				return l
+			}
+			l = i + 1
+		}
+	}
+	if len(t) == len(s)-l && strings.Compare(s[l:], t) == 0 {
+		return l
+	}
+	return -1
+}
+
+// SimpleCases captures simple SQL statements and uses string replacement instead of `restore` to improve performance.
+// See https://github.com/pingcap/tidb/issues/22398.
+func SimpleCases(node ast.StmtNode, defaultDB, origin string) (s string, ok bool) {
+	if len(origin) == 0 {
+		return "", false
+	}
+	insert, ok := node.(*ast.InsertStmt)
+	if !ok {
+		return "", false
+	}
+	if insert.Select != nil || insert.Setlist != nil || insert.OnDuplicate != nil || (insert.TableHints != nil && len(insert.TableHints) != 0) {
+		return "", false
+	}
+	join := insert.Table.TableRefs
+	if join.Tp != 0 || join.Right != nil {
+		return "", false
+	}
+	ts, ok := join.Left.(*ast.TableSource)
+	if !ok {
+		return "", false
+	}
+	tn, ok := ts.Source.(*ast.TableName)
+	if !ok {
+		return "", false
+	}
+	parenPos := strings.Index(origin, "(")
+	if parenPos == -1 {
+		return "", false
+	}
+	if strings.Contains(origin[:parenPos], ".") {
```

Comment:
Is it possible that the origin SQL contain comments like `insert /*+ somefunc(t.col) */ ...`?

Reply:
I have filtered out this situation， you can see above.
```golang
insert.TableHints != nil && len(insert.TableHints) != 0
```